activesupport/lib/active_support/inflector/transliterate.rb

```diff
@@ -75,13 +75,21 @@ def parameterize(string, sep = '-')
       # Turn unwanted chars into the separator
       parameterized_string.gsub!(/[^a-z0-9\-_]+/i, sep)
       unless sep.nil? || sep.empty?
-        re_sep = Regexp.escape(sep)
+        if sep == "-".freeze
```

Comment:
Not sure what is wrong. Are you  concerned that `"-".freeze` does not == `"-"` ? It does

```
puts "-".freeze == "-"
true
puts "-" == "-".freeze
true
```